src/axelar/AxelarTransceiver.sol

```diff
@@ -0,0 +1,182 @@
+// Use their libraries
+import {Transceiver} from "@wormhole-foundation/native_token_transfer/Transceiver/Transceiver.sol";
+
+import {IAxelarTransceiver} from "./interfaces/IAxelarTransceiver.sol";
+
+contract AxelarTransceiver is IAxelarTransceiver, AxelarGMPExecutable, Transceiver {
+    IAxelarGasService public immutable gasService;
+
+    // These mappings are used to convert chainId and chainName between Wormhole and Axelar formats.
+    struct AxelarTransceiverStorage {
+        mapping(uint16 => string) idToAxelarChainId;
+        mapping(string => uint16) axelarChainIdToId;
+        mapping(uint16 => string) idToTransceiverAddress;
+        mapping(string => uint16) transceiverAddressToId;
+    }
+    // keccak256(abi.encode(uint256(keccak256("openzeppelin.storage.AxelarTransceiver")) - 1)) & ~bytes32(uint256(0xff))
+
+    bytes32 internal constant AXELAR_TRANSCEIVER_STORAGE_SLOT =
+        0x6d72a7741b755e11bdb1cef6ed3f290bbe196e69da228a3ae322e5bc37ea7600;
+
+    // TODO: update this based on tests
+    uint256 internal constant DESTINATION_EXECUTION_GAS_LIMIT = 200000;
+
+    error UnsupportedMessageType();
+    error InvalidSibling(uint16 chainId, string sourceChain, string sourceAddress);
+    error NotImplemented();
+
+    constructor(
+        address _gateway,
+        address _gasService,
+        address _manager
+    ) AxelarGMPExecutable(_gateway) Transceiver(_manager) {
+        gasService = IAxelarGasService(_gasService);
+    }
+
+    /**
+     * Set the bridge manager contract address
+     * @param chainId The chainId of the chain. This is used to identify the chain in the EndpointManager.
+     * @param chainName The chainName of the chain. This is used to identify the chain in the AxelarGateway.
+     * @param transceiverAddress The address of the tranceiver on the other chain, in the axelar accepted format.
+     */
+    function setAxelarChainId(
+        uint16 chainId,
+        string calldata chainName,
+        string calldata transceiverAddress
+    ) external onlyOwner {
+        AxelarTransceiverStorage storage slot = _storage();
+        slot.idToAxelarChainId[chainId] = chainName;
+        slot.axelarChainIdToId[chainName] = chainId;
+        slot.idToTransceiverAddress[chainId] = transceiverAddress;
+        slot.transceiverAddressToId[transceiverAddress] = chainId;
+    }
+
+    /// @notice Fetch the delivery price for a given recipient chain transfer.
+    /// @param recipientChainId The Wormhole chain ID of the target chain.
+    /// param instruction An additional Instruction provided by the Transceiver to be
+    ///        executed on the recipient chain.
+    /// @return deliveryPrice The cost of delivering a message to the recipient chain,
+    ///         in this chain's native token.
+    function _quoteDeliveryPrice(
+        uint16 recipientChainId,
+        TransceiverStructs.TransceiverInstruction memory /*instruction*/
+    ) internal view virtual override returns (uint256) {
```

Comment:
No need to be decorated with `virtual`? Same goes for `_sendMessage` and `_callContract`

Reply:
it is nice in case we want a contract that makes a small change on this in the future, and does no harm otherwise.